src/plugins/package_compiler_lib.jl

```diff
@@ -0,0 +1,119 @@
+using PkgTemplates: @plugin, @with_kw_noshow, Plugin
+
+camel_to_snake_case(str::AbstractString) = replace(str, r"([a-z])([A-Z]+)" => s"\1_\2") |> lowercase
+
+"""
+    PackageCompilerLib(;
+        lib_name=nothing,
+        build_jl="$(contractuser(default_file("build", "build.jl")))",
+        generate_precompile_jl="$(contractuser(default_file("build", "generate_precompile.jl")))",
+        additional_precompile_jl="$(contractuser(default_file("build", "additional_precompile.jl")))",
+        install_sh="$(contractuser(default_file("build", "install.sh")))",
+        install_txt="$(contractuser(default_file("build", "INSTALL.txt")))",
+        lib_h="$(contractuser(default_file("build", "lib.h")))",
+        project_toml="$(contractuser(default_file("build", "Project.toml")))",
+        makefile="$(contractuser(default_file("Makefile")))",
+        additional_gitignore=[],
+    )
+
+Adds files which facilitate the creation of a C-library from the generated project.
+
+See [PackageCompiler.jl](https://github.com/JuliaLang/PackageCompiler.jl) for more information.
+
+## Keyword Arguments
+- `lib_name::Union{String, Nothing}`: Name of the library to generate.  If `nothing`,
+  defaults to the snake-case version of the package name.
+- `build_jl::String`: The file used to generate the C library.  Calls out to `PackageCompiler.jl`.
```

Comment:
```suggestion
- `lib_name::Union{String, Nothing}`: Name of the library to generate. If `nothing`,
  defaults to the snake-case version of the package name.
- `build_jl::String`: The file used to generate the C library. Calls out to `PackageCompiler.jl`.
```
Just removes extra whitespace